#! /usr/bin/env python

PACKAGE='flyer_controller'
import roslib; roslib.load_manifest(PACKAGE)

from driver_base.msg import SensorLevels
from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()
RUNNING = SensorLevels.RECONFIGURE_RUNNING
command_frame_enum = gen.enum( [ gen.const("internal", str_t, "internal", "internal frame"),
                                 gen.const("external", str_t, "external", "external frame") ], "Enum to set the command frame.")
# Name, Type, Reconfiguration level, Description, Default, Min, Max
gen.add("KP", double_t, RUNNING, "Proportional gain", 15.0, 0.0, 30.0)
gen.add("KI", double_t, RUNNING, "Integral gain", 1.0, 0.0, 2.0)
gen.add("KD", double_t, RUNNING, "Derivative gain", 15.0, 0.0, 30.0)
gen.add("command_frame", str_t, RUNNING, "Command frame (internal/external)", "internal", edit_method = command_frame_enum)
gen.add("external_frame_heading", double_t, RUNNING, "Heading [deg] that external command frame points to", 0.0, 0.0, 359.0)

exit(gen.generate(PACKAGE, "controller.py", "controller"))

